src/lib.rs

```diff
@@ -1390,20 +1393,42 @@ pub mod rpc {
             })
         }
 
-        fn closed(&mut self) -> Pin<Box<dyn Future<Output = ()> + Send + '_>> {
+        fn closed(&mut self) -> Pin<Box<dyn Future<Output = ()> + Send + Sync + '_>> {
             Box::pin(async move {
                 self.send.stopped().await.ok();
             })
         }
+    }
 
-        fn is_rpc(&self) -> bool {
-            true
+    struct QuinnSender<T>(tokio::sync::Mutex<QuinnSenderInner<T>>);
+
+    impl<T> Debug for QuinnSender<T> {
+        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+            f.debug_struct("QuinnSender").finish()
         }
     }
 
-    impl<T> Drop for QuinnSender<T> {
```

Comment:
This is not needed, the underlying `quinn::SendStream` has a `Drop` impl that also `finish`es the stream.